python_modules/dagster/dagster_tests/core_tests/execution_tests/test_instance_concurrency_context.py

```diff
@@ -141,32 +157,81 @@ def test_custom_interval(concurrency_custom_sleep_instance):
 
 
 def test_unset_concurrency_default(concurrency_instance):
-    run_id = make_new_run_id()
+    run = concurrency_instance.create_run_for_job(define_foo_job(), run_id=make_new_run_id())
 
     assert concurrency_instance.event_log_storage.get_concurrency_keys() == set()
 
-    with InstanceConcurrencyContext(concurrency_instance, run_id) as context:
+    with InstanceConcurrencyContext(concurrency_instance, run) as context:
         assert context.claim("foo", "a")
         assert context.claim("foo", "b")
 
 
 def test_default_concurrency_key(concurrency_instance_with_default_one):
-    run_id = make_new_run_id()
+    run = concurrency_instance_with_default_one.create_run_for_job(
+        define_foo_job(), run_id=make_new_run_id()
+    )
 
     assert concurrency_instance_with_default_one.event_log_storage.get_concurrency_keys() == set()
     assert concurrency_instance_with_default_one.global_op_concurrency_default_limit == 1
 
-    with InstanceConcurrencyContext(concurrency_instance_with_default_one, run_id) as context:
+    with InstanceConcurrencyContext(concurrency_instance_with_default_one, run) as context:
         assert context.claim("foo", "a")
         assert not context.claim("foo", "b")
 
 
 def test_zero_concurrency_key(concurrency_instance):
-    run_id = make_new_run_id()
+    run = concurrency_instance.create_run_for_job(define_foo_job(), run_id=make_new_run_id())
 
     concurrency_instance.event_log_storage.set_concurrency_slots("foo", 0)
     assert concurrency_instance.event_log_storage.get_concurrency_keys() == set(["foo"])
 
-    with InstanceConcurrencyContext(concurrency_instance, run_id) as context:
+    with InstanceConcurrencyContext(concurrency_instance, run) as context:
         assert not context.claim("foo", "a")
         assert not context.claim("foo", "b")
+
+
+def test_step_priority(concurrency_instance):
+    run = concurrency_instance.create_run_for_job(define_foo_job(), run_id=make_new_run_id())
+    concurrency_instance.event_log_storage.set_concurrency_slots("foo", 1)
+    with InstanceConcurrencyContext(concurrency_instance, run) as context:
+        assert context.claim("foo", "a")
+        assert not context.claim("foo", "b")
+        assert not context.claim("foo", "c", step_priority=1000)
+        assert not context.claim("foo", "d", step_priority=-1)
+
+        context.free_step("a")
+        time.sleep(1)
+
+        # highest priority step should be able to claim the slot
+        assert not context.claim("foo", "b")
+        assert context.claim("foo", "c")
+        assert not context.claim("foo", "d")
+
+
```

Comment:
should there be a test case for step + run? A case where one is positive and one is negative could be nice, like a static job with step priority `X` but one of the dagster run is "backgrounded" with priority `-Y`

Reply:
will add a test